cmd/deployment-check/input.go

```diff
@@ -125,6 +125,18 @@ func parseInputValues() {
 		log.Infoln("Parsed CHECK_DEPLOYMENT_REPLICAS:", checkDeploymentReplicas)
 	}
 
+	// Parse incoming deployment node selectors
+	if len(checkDeploymentNodeSelectorsEnv) != 0 {
+		splitEnvVars := strings.Split(checkDeploymentNodeSelectorsEnv, ",")
+		for _, splitEnvVarKeyValuePair := range splitEnvVars {
+			parsedEnvVarKeyValuePair := strings.Split(splitEnvVarKeyValuePair, "=")
+			if _, ok := checkDeploymentNodeSelectors[parsedEnvVarKeyValuePair[0]]; !ok {
```

Comment:
Why do we check for the key to exist here rather than just setting the key/value pair we want?

Also, I would add a couple len checks to ensure that the len of the things you're splitting is `2`.  If it's not, you continue and log an error because something isn't balanced.

Reply:
whatever goes here also goes on the daemonset-check code...

Reply:
I first check for the key to exist incase a user inputs something similar to `foo=bar,foo=baz` -- in this case, should I check for a len of `3`? `a=x` has three chars because of the equals sign

Reply:
yeah, 3 for the length of string makes perfect sense there